### DISCUSSION TITLE
Component Props

### DISCUSSION THREAD

**Discussion by sutaungmai** [huntabyte/shadcn-svelte#1539] [Help] [answered]
I got some unexpected error. Here is my index.ts of select component:

![code](https://github.com/user-attachments/assets/1d370a7d-9f6a-4696-9afe-4625b0d03db2)

And here is my usage:

![code](https://github.com/user-attachments/assets/eb2f8c04-1ddc-4463-8400-b8f81a06c7e6)

But I got error when giving id or other props. Here is its error message:

![image](https://github.com/user-attachments/assets/a7a18078-d63a-4cfa-b4c0-441047578cd1)

**Comment by huntabyte** [comment]
You should pass the id and name to the `<SelectTrigger>` not the `<Select>`

**Reply by sutaungmai** [reply]
It seems Select component isn't accepting props. I move `id` to `<SelectTrigger>` but I got the same error again on type. Is there anything that I need to declare type declaration or something? I am using bits-ui version `1.0.0-next.69`. And also I got this kind of similar error on all of the other select related components.
![image](https://github.com/user-attachments/assets/8dea5e2b-1b61-4bb3-85f1-784d2c7291d5)

**Reply by sutaungmai** [reply, marked as answer]
Aha I got it, the problem could be that I am using `npm` inside `pnpm` project. But I solved it by re-initialting the process. Thanks for the help!! :)
